analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -183,6 +185,31 @@ open class Npm(
         }
     }
 
+    /**
+     * Parse information about the author. According to
+     * https://docs.npmjs.com/files/package.json#people-fields-author-contributors, there are two formats to
+     * specify the author of a package: An object with multiple properties or a single string.
+     */
+    private fun parseAuthor(json: JsonNode): SortedSet<String> =
+        sortedSetOf<String>().apply {
+            json["author"]?.let { authorNode ->
+                when {
+                    authorNode.isObject -> authorNode["name"]?.textValue()
+                    authorNode.isTextual -> parseAuthorString(authorNode.textValue())
+                    else -> null
+                }
+            }?.let { add(it) }
+        }
+
+    /**
+     * Parse the author if it is defined as a single string in the format "Name <mail> (url)". Try to extract the name
+     * part from the string or return the full string if no mail or url components are found.
+     */
+    private fun parseAuthorString(authorStr: String?): String? = authorStr?.let { str ->
+        val posSeparator = min(str.findSeparator('<'), str.findSeparator('('))
```

Comment:
How about instead leveraging `substringBefore()`, or maybe better yet, `split('<', '(', limit = 2)`, to be able to get rid of the `findSeparator` helpfer function?

Reply:
Implemented using `substringBefore()`.